sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -757,7 +757,10 @@ class StreamSuite extends StreamTest {
 
         QueryTest.checkAnswer(spark.table("counts").toDF(),
           Row("1", 1) :: Row("2", 1) :: Row("3", 2) :: Row("4", 2) ::
-          Row("5", 2) :: Row("6", 2) :: Row("7", 1) :: Row("8", 1) :: Row("9", 1) :: Nil)
+          Row("5", 2) :: Row("6", 2) :: Row("7", 1) :: Row("8", 1) :: Row("9", 1) :: Nil) match {
+          case Some(errorMessage) => fail(errorMessage)
+          case None =>
+        }
```

Comment:
This seems to fail like the following, but the result look correct. Did I miss something?
```
!== Correct Answer - 9 ==   == Spark Answer - 9 ==
!struct<>                   struct<value:int,count(1):bigint>
 [1,1]                      [1,1]
 [2,1]                      [2,1]
 [3,2]                      [3,2]
 [4,2]                      [4,2]
 [5,2]                      [5,2]
 [6,2]                      [6,2]
 [7,1]                      [7,1]
 [8,1]                      [8,1]
 [9,1]                      [9,1]
```

Reply:
This is because `expectedAnswer`'s data type do not match. https://github.com/apache/spark/pull/26768/commits/61eacc940381ef717f102c809ef80afd9b0facf7 fix this issue.

Reply:
Before this PR. it will not fail.

Reply:
Thanks!